packages/agent/src/store-managed-did.ts

```diff
@@ -0,0 +1,299 @@
+    const { agent, alias, context, did } = options;
+
+    // Determine which DID to use to author DWN messages.
+    const authorDid = await this.getAuthor({ agent, context, did });
+
+    // Query the DWN for all stored DID objects.
+    const { reply: queryReply} = await agent.dwnManager.processRequest({
+      author         : authorDid,
+      target         : authorDid,
+      messageType    : 'RecordsQuery',
+      messageOptions : {
+        filter: { ...this._didRecordProperties }
+      }
+    });
+
+    // Loop through all of the entries and return a match, if found.
+    for (const record of queryReply.entries ?? []) {
+      if (record.encodedData) {
+        const storedDid = Convert.base64Url(record.encodedData).toObject() as ManagedDid;
+        if (storedDid && storedDid.did === did) return storedDid;
+        if (storedDid && storedDid.alias === alias) return storedDid;
+      }
+    }
+
+    // Return undefined if no matches were found.
+    return undefined;
+  }
+
+  async getDid(options: {
+    agent: Web5ManagedAgent,
+    context?: string,
+    did: string
+  }): Promise<ManagedDid | undefined> {
+    const { agent, context, did } = options;
+
+    // Determine which DID to use to author DWN messages.
+    const authorDid = await this.getAuthor({ agent, context, did });
+
+    // Query the DWN for all stored DID objects.
+    const { reply: queryReply} = await agent.dwnManager.processRequest({
+      author         : authorDid,
+      target         : authorDid,
+      messageType    : 'RecordsQuery',
+      messageOptions : { filter: { ...this._didRecordProperties } }
+    });
+
+    // Loop through all of the entries and return a match, if found.
+    for (const record of queryReply.entries ?? []) {
+      if (record.encodedData) {
+        const storedDid = Convert.base64Url(record.encodedData).toObject() as ManagedDid;
+        if (storedDid && storedDid.did === did) return storedDid;
+      }
+    }
+
+    // Return undefined if no matches were found.
+    return undefined;
+  }
+
+  async importDid(options: {
+    agent: Web5ManagedAgent,
+    context?: string,
+    did: ManagedDid
+  }) {
+    const { agent, context, did: importDid } = options;
+
+    // Determine which DID to use to author DWN messages.
+    const authorDid = await this.getAuthor({ agent, context, did: importDid.did });
+
+    // Check if the DID being imported is already present in the store.
+    const duplicateFound = await this.getDid({ agent, context, did: importDid.did });
+    if (duplicateFound) {
+      throw new Error(`DidStoreDwn: DID with ID already exists: '${importDid.did}'`);
+    }
+
+    // Encode the ManagedDid as bytes.
+    const importDidU8A = Convert.object(importDid).toUint8Array();
+
+    const writeReply = await agent.dwnManager.processRequest({
+    // const { reply: { status } } = await agent.dwnManager.processRequest({
+      author         : authorDid,
+      target         : authorDid,
```

Comment:
target will always be author did here?

Reply:
Yes, but `authorDid` is set conditionally here:
https://github.com/TBD54566975/web5-js/pull/168/files/1d18158a0b1f154afba713af994e2206aa403169#diff-b1f15630c70e816fd04cdc7ea2fb47e3fd5e42fb51a7eaf167029351adee05b0R147